WebDriverAgentLib/Commands/FBElementCommands.m

```diff
@@ -629,6 +629,21 @@ + (NSArray *)routes
     isSuccessful = [element fb_selectNextOptionWithOffset:offset error:&error];
   } else if ([order isEqualToString:@"previous"]) {
     isSuccessful = [element fb_selectPreviousOptionWithOffset:offset error:&error];
+  } else if (request.arguments[@"value"]) {
+    NSString *value = request.arguments[@"value"];
+    NSString* endValue = element.wdValue;
+    if(element.wdValue == value) {
+      return FBResponseWithOK();
+    }
+    do {
+      [element fb_selectNextOptionWithOffset:offset error:&error];
+      element = [elementCache elementForUUID:(NSString *)request.parameters[@"uuid"]];
+      if([element.wdValue isEqualToString: value]) {
+        return FBResponseWithOK();
+      }
+    } while(![element.wdValue isEqualToString: endValue]);
```

Comment:
this might be an endless loop if the target value does not exist or cannot be reached. We must always provide either a timeout or the number of maximum attempts

Reply:
okay i will try putting the timeout

Reply:
or should i add a Maximum attempts option, which user may change from settings?

Reply:
So should i add a new setting for user to change?

Reply:
yes, you could add a new arg, for example maxAttempts and just assign it the default value if not provired